### ISSUE TITLE
Address warning from `jsonschema` 4.18.x

### ISSUE THREAD

**Issue by allanlewis** [python-jsonschema/hypothesis-jsonschema#102]
As of v4.18.0, `jsonschema` emits the following warning related to `jsonschema.exceptions.RefResolutionError`:

```
DeprecationWarning: jsonschema.exceptions.RefResolutionError is deprecated as of version 4.18.0. If you wish to catch potential reference resolution errors, directly catch referencing.exceptions.Unresolvable.
```

It would be good if this package could address this warning before `RefResolutionError` is removed in some future version.

**Comment by allanlewis** [comment]
In ignoring this warning in my project (I set `PYTHONWARNINGS=error` when running unit tests) I then get another warning:

```
DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
```

So there might be a chain of deprecations to handle here 😕

**Comment by allanlewis** [comment]
These changes are documented in the release notes for `jsonschema` 4.18.0: https://github.com/python-jsonschema/jsonschema/releases/tag/v4.18.0
